--- a/include/linux/phy.h
+++ b/include/linux/phy.h
@@ -879,6 +879,14 @@ struct phy_driver {
 	int (*get_sqi)(struct phy_device *dev);
 	/** @get_sqi_max: Get the maximum signal quality indication */
 	int (*get_sqi_max)(struct phy_device *dev);
+	/** @get_port: Get port */
+	int (*get_port)(struct phy_device *dev);
+	/** @set_port: Set port */
+	int (*set_port)(struct phy_device *dev, int port);
+	/** @get_eee: Get Energy Efficient Ethernet */
+	int (*get_eee)(struct phy_device *dev, struct ethtool_eee *e);
+	/** @set_eee: Set Energy Efficient Ethernet */
+	int (*set_eee)(struct phy_device *dev, struct ethtool_eee *e);
 };
 #define to_phy_driver(d) container_of(to_mdio_common_driver(d),		\
 				      struct phy_driver, mdiodrv)
